conda/core/python_dist.py

```diff
@@ -0,0 +1,444 @@
+    elif ".dist-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonInstalledDistribution
+        package_type = PackageType.SHADOW_PYTHON_DIST_INFO
+    elif anchor_file.endswith(".egg-info"):
+        sp_reference = basename(anchor_file)
+        dist_file = join(prefix_path, win_path_ok(anchor_file))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_FILE
+    elif ".egg-info" in anchor_file:
+        sp_reference = basename(dirname(anchor_file))
+        dist_file = join(prefix_path, win_path_ok(dirname(anchor_file)))
+        dist_cls = PythonEggInfoDistribution
+        package_type = PackageType.SHADOW_PYTHON_EGG_INFO_DIR
+    else:
+        raise NotImplementedError()
+
+    try:
+        pydist = dist_cls(dist_file)
+    except MetadataConflictError:
+        print("MetadataConflictError:", anchor_file)
+        pydist = None
+
+    return pydist, sp_reference, package_type
+
+
+def get_python_record(anchor_file, prefix_path):
+    """"""
+    pydist, sp_reference, package_type = get_python_distribution_info(anchor_file, prefix_path)
+
+    if pydist is None:
+        return None
+
+    # x.provides  =>  [u'skdata (0.0.4)']
+    # x.run_requires  =>  set([u'joblib', u'scikit-learn', u'lockfile', u'numpy', u'nose (>=1.0)'])  # NOQA
+    # >>> list(x.list_installed_files())  =>  [(u'skdata/__init__.py', u'sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU', u'0'), (u'skdata/base.py', u'sha256=04MW02dky5T4nZb6Q0M351aRbAwLxd8voCK3nrAU-g0', u'5019'), (u'skdata/brodatz.py', u'sha256=NIPWLawJ59Fr037r0oT_gHe46WCo3UivuQ-cwxRU3ow', u'8492'), (u'skdata/caltech.py', u'sha256=cIfyMMRYggZ3Jkgc15tsYi_ZsZ7NpRqWh7mZ8bl6Fo0', u'8047'), (u'skdata/data_home.py', u'sha256=o5ChOI4v3Jd16JM3qWZlhrs5q-g_0yKa5-Oq44HC_K4', u'1297'), (u'skdata/diabetes.py', u'sha256=ny5Ihpc_eiIRYgzFn3Lm81fV0SZ1nyZQnqEmwb2PrS0', u'995'), (u'skdata/digits.py', u'sha256=DipeWAb3APpjXfmKmSumkfEFzuBW8XJ0  # NOQA
+
+    # TODO: normalize names against '.', '-', '_'
+    # TODO: ensure that this dist is *actually* the dist that matches conda-meta
+
+    channel = Channel('pypi')
+    if package_type == PackageType.SHADOW_PYTHON_EGG_INFO_FILE:
+        paths_data = None
+    elif package_type in (PackageType.SHADOW_PYTHON_DIST_INFO,
+                          PackageType.SHADOW_PYTHON_EGG_INFO_DIR):
+        paths_data = pydist.get_prefix_records()
+    elif package_type == PackageType.SHADOW_PYTHON_EGG_LINK:
+        paths_data = pydist.get_prefix_records()
+        channel = Channel('<develop>')
+    else:
+        raise NotImplementedError()
+
+    # TODO: need to add entry points, "exports," and other files that might not be in RECORD  # NOQA
+
+    depends = pydist.get_run_requirements(exclude_markers=True)  # For now...
+
+    # TODO: need to add python (with version?) to deps
+
```

Comment:
Any of the TODOs you would like me to work on this PR @kalefranz ?

Reply:
Go ahead and fill in what you think has an obvious answer.

Reply:
This one is yes, we do.  With version. :)

Reply:
👍